gnark-prover/server/server.go

```diff
@@ -122,39 +111,31 @@ func (handler inclusionHandler) ServeHTTP(w http.ResponseWriter, r *http.Request
 		return
 	}
 
-	var proof *prover.Proof
+	var circuitType prover.CircuitType
 
-	var params prover.InclusionParameters
-
-	err = json.Unmarshal(buf, &params)
+	circuitType, err = prover.ParseCircuitType(buf)
```

Comment:
why don't we just use t[he specified input Json ](https://gist.github.com/sergeytimoshin/beec2246537a30bd5adb1b87b51ba893)here and decide why circuit to use based on that?
```

if new addresses and input compressed accounts -> combined
else if input compressed accounts -> inclusion
else if new addresses -> non inclusion
```

Reply:
It's basically what we are doing there. :)

What I don't really like is that I still have ["inclusion" & "non-inclusion" flags for cli](https://github.com/Lightprotocol/light-protocol/pull/600/files/2c2c17017cd05c8fc82d77cfc55edcebe4cf5ee4#diff-c39d39cf374913058da43546ec23bde4639faf0e7619cd897823d21cff6d5bb9L45-L46) cmds, it can be confusing.
But it allows us to start server with only needed circuits => optimise CI execution.